tests/CollectionTests.cpp

```diff
@@ -680,7 +709,8 @@ TEST_F(CollectionTest##s##w##e##o, Random) { random(); } \
 TEST_F(CollectionTest##s##w##e##o, RandomOutOfRange) { random(); outOfRange(); } \
 TEST_F(CollectionTest##s##w##e##o, RandomMultiThreaded) { randomMultiThreaded(); } \
 TEST_F(CollectionTest##s##w##e##o, List) { list(0.1); } \
-TEST_F(CollectionTest##s##w##e##o, BinarySearch) { binarySearch(); }
+TEST_F(CollectionTest##s##w##e##o, BinarySearch) { binarySearch(); } \
+TEST_F(CollectionTest##s##w##e##o, BinarySearchNotFound) { binarySearch_notFound(); }
```

Comment:
can we also add some high level "integration" test with the evidence containing non-existant profile

Reply:
let's add them as part of `device-detection-cxx`